http/dotnet/httpclient/src/HttpClientBuilder.cs

```diff
@@ -0,0 +1,48 @@
+using System.Linq;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Kiota.Abstractions;
+
+namespace Microsoft.Kiota.Http.HttpClient {
```

Comment:
@baywet, I think @nikithauc is referring to this.

Reply:
aaah we had a long naming debate with @darrelmiller a long time ago.
I wanted to name this `Microsoft.Kiota.Transport.Http` leaving room for other transport implementations potentially in the future. Thoughts on that name? And for typescript we'd have something like `@microsoft/kiota-transport-http`.

Reply:
What are transport implementations?

Reply:
well one of them is HTTP of course, another one could be gRPC, etc...

Reply:
I think another common thing between HTTP and gRPC is that they are both protocols. So we can also have `Microsoft.Kiota.Protocol.Http` or `Microsoft.Kiota.Protocols.Http` as the namespace alternatives. I am OK with `Microsoft.Kiota.Transport.Http` too.

Reply:
I don't think transport is correct here. Protocol is more accurate. For example, if we introduced WebSockets, we'd have:
`Microsoft.Kiota.Protocol.Http`
`Microsoft.Kiota.Protocol.WebSocket`

They use the same transport.

Reply:
`Microsoft.Kiota.Protocol.Http`
`Microsoft.Kiota.Protocol.WebSocket` 

The namespace looks meaningful. 

> Hypertext Transfer Protocol (HTTP) is an application-layer protocol for transmitting hypermedia documents, such as HTML. It was designed for communication between web browsers and web servers, but it can also be used for other purposes.

https://developer.mozilla.org/en-US/docs/Web/HTTP

Reply:
yep, protocol works for me as well. As this is a discussion on existing naming pattern, and as I'd like Darrel's "blessing" before changing the naming, I'm going to create a separate issue for that. Thanks for all the great suggestions!

Reply:
#444